### PR TITLE
feat(Select): Use Radix Select for Select

### PR DESCRIPTION
**Pull request by okonet**
This PR introduces Radix Select instead of the old styles Select. This is still WIP since we should replace all `Menu` usages with it it seems.

### CODE REVIEW

**Review by kof — APPROVED** [review]

**Comment by kof** [comment]
> A subtle quirk in the interaction of the select.

this is super nice, that's why you don't build this type of stuff ad-hoc from scratch every time

**Comment by okonet** [comment]
Yeah I also noticed some weirdness in their implementation. I'll have to explore the combo box behavior before settling on the final solution though. 

It might not work with this Radix primitive so I'm still in the exploration stage unfortunately. This stuff is quite complex and I wish The Platform would simple give us an ability to style selects and comboboxes easily without doing this kind of work. Maybe in a decade... 🤔

**Comment by okonet** [comment]
I updated styles to match the theme and it looks much better now. Also found the reason for wrong positioning on the dashboard page. I think it's ready for the review and we could merge this for now. It doesn't support combo box behavior and probably wont but it's good for most use cases.

**Comment by kof** [comment]
cool fixing the ts error and merging now

File: app/shared/design-system/components/select.tsx [hunk 1]
```diff
@@ -31,44 +39,161 @@ const SelectWrapper = styled("div", {
...
+  SelectPrimitive.ScrollUpButton,
+  scrollButtonStyles
+);
+
+const SelectScrollDownButton = styled(
+  SelectPrimitive.ScrollDownButton,
+  scrollButtonStyles
+);
+
+const SelectItemBase = ({ children, ...props }: SelectItemProps) => {
```
[the remark this answers is not in the document]
Comment (kof) [reply app/shared/design-system/components/select.tsx]: done

**Comment by kof** [comment]
merged 🎉

**Pull request merged**